config/config.exs

```diff
@@ -22,6 +31,24 @@ config :logger, :console,
   format: "$time $metadata[$level] $message\n",
   metadata: [:request_id]
 
+git_branch =
+  Config.get_env("GIT_COMMIT") ||
+    System.cmd("git", ~w(rev-parse --abbrev-ref HEAD))
+    |> elem(0)
+    |> String.trim
+
+git_commit =
+  Config.get_env("GIT_COMMIT") ||
+    System.cmd("git", ~w(rev-parse HEAD))
```

Comment:
This crashes compilation on systems without `git`.

I did it like this once before: [Status.System.commit/0](https://github.com/shinyscorpion/elixir-status/blob/prototype/pre0/lib/status/system.ex#L5) and is now used in multiple projects.

We can do something similar or we can include the `:status` package and I'll add `branch/0` to that.

Reply:
I'm ok with the crashes, our dev machines will have it, and at build time it should fail if we haven't set the environment variable